### PR TITLE
New command Convert-JsonToBicep

### PR DESCRIPTION
**Pull request by StefanIvemo**
A first draft to fix feature request #105.

To use the feature:
```
$json = @"
[
  {
    "properties": {
      "NSGName": "subnet2-nsg",
      "SubnetName": "subnet2",
      "RouteName": "",
      "disableBgpRoutePropagation": true,
      "routes": []
    }
  },
  {
    "properties": {
      "NSGName": "subnet3-nsg",
      "SubnetName": "subnet3",
      "RouteName": "",
      "disableBgpRoutePropagation": false,
      "routes": []
    }
  }
]
"@
Convert-JsonToBicep -String $jsonSnippet
```
Or:
`Convert-JsonToBicep -String '{"key": "value", "anotherKey": "anotherValue"}'`

### CODE REVIEW

**Comment by PalmEmanuel** [comment]
I'm not sure I like the name "ARMSnippet" for the parameter. I suggest we simply name it "String" instead.

`Convertto-Bicep -String $jsonSnippet`

**Comment by StefanIvemo** [comment]
Much better @PalmEmanuel. I was searching for a better name for it.

File: Source/Public/Convert-JsonToBicep.ps1 [hunk 1]
```diff
@@ -0,0 +1,41 @@
+function Convert-JsonToBicep {
+    [CmdletBinding()]
+    param(
+        [Parameter(Mandatory, 
+            ParameterSetName = 'String')]
+        [ValidateNotNullOrEmpty()]
+        [ValidateScript({$_ | Convertfrom-Json})]
```
Comment (bjompen) [inline Source/Public/Convert-JsonToBicep.ps1:7]: Add a try / catch block to improve error message. 
```PowerShell
[ValidateScript({ 
  try { $_ | Convertfrom-Json }
  catch {throw 'not a valid json'}
})]
```

Error as it looks now:
```PowerShell
> Convert-JsonToBicep -String 'NotAJson'
Convert-JsonToBicep: Cannot validate argument on parameter 'String'. Conversion from JSON failed with error: Error parsing NaN value. Path '', line 1, position 1.
```

with catch:
```PowerShell
> Convert-JsonToBicep -String 'NotAJson'
Convert-JsonToBicep: Cannot validate argument on parameter 'String'. not a valid json
```

File: Source/Public/Convert-JsonToBicep.ps1 [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+        $jsonObject = ConvertFrom-Json -InputObject $String -AsHashtable -Depth 100
+        $variables = [ordered]@{}
+        $templateBase = [ordered]@{
+            '$schema'        = 'https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#'
+            'contentVersion' = '1.0.0.0'
+        }
+        $variables['temp'] = $jsonObject                                  
+        $templateBase['variables'] = $variables
+        $tempTemplate = ConvertTo-Json -InputObject $templateBase -Depth 100
+        Out-File -InputObject $tempTemplate -FilePath "$($env:TEMP)\tempfile.json"
```
Comment (bjompen) [inline Source/Public/Convert-JsonToBicep.ps1:26]: If we need to create a tempfile, we should clean up after the process is done. Please add Remove-Item after the file has been used.

**Review by SimonWahlin — APPROVED** [review]
LGTM

**Pull request merged**